frontend/src/metabase/nav/components/SearchResults.tsx

```diff
@@ -0,0 +1,113 @@
+import type { LegacyRef } from "react";
+import { useEffect, useMemo, useState } from "react";
+import { t } from "ttag";
+import { push } from "react-router-redux";
+import { useListKeyboardNavigation } from "metabase/hooks/use-list-keyboard-navigation";
+import { SearchResult } from "metabase/search/components/SearchResult";
+import { EmptyStateContainer } from "metabase/nav/components/SearchResults.styled";
+import EmptyState from "metabase/components/EmptyState";
+import type { SearchFilters } from "metabase/search/types";
+import {
+  DEFAULT_SEARCH_LIMIT,
+  SEARCH_DEBOUNCE_DURATION,
+} from "metabase/lib/constants";
+import { useSearchListQuery } from "metabase/common/hooks";
+import { useDispatch } from "metabase/lib/redux";
+import Search from "metabase/entities/search";
+import { Loader, Text, Stack } from "metabase/ui";
+import type { SearchModelType } from "metabase-types/api";
+import { useDebouncedEffect } from "metabase/hooks/use-debounced-effect";
+
+type SearchResultsProps = {
+  onEntitySelect?: (result: any) => void;
```

Comment:
`WrappedResult`?

Reply:
There's another usage in `LinkViz` that takes in `UnrestrictedLinkEntity`. I didn't want to touch it to keep this refactor small and we could handle that in a separate PR, but I can look a little more and see if there's a way to make it work